tools/lsp/preview/drop_location.rs

```diff
@@ -1125,18 +1125,10 @@ pub fn create_move_element_workspace_edit(
             let (path, selection_offset) = element.path_and_offset();
             return Some((edit, DropData { selection_offset, path }));
         } else {
-            let children = drop_info.target_element_node.children();
-            let child_index = {
-                let tmp =
-                    children.iter().position(|c| c == element).expect("We have the same parent");
-                if tmp == children.len() {
-                    usize::MAX
-                } else {
-                    tmp
-                }
-            };
-
-            if child_index == drop_info.child_index {
+            let children = &drop_info.target_element_node.children();
+            if children.iter().position(|c| c == element).unwrap_or(usize::MAX - 1)
+                == drop_info.child_index
+            {
```

Comment:
Would that work?

```suggestion
            let children = &drop_info.target_element_node.children();
            if children.get(drop_info.child_index).is_some_and(|c| c == element) {
```

Then you don't need to do a linear search in the children.
Note that this is going to have a different result if drop_info.child_index == `usize::MAX - 1` and that the element is not in the children list, but that doesn't seem intended since you write in the commit message it shouldn't happen.

Also the code before used `usize::MAX` and then uses `usize::MAX - 1` so it seems that this is indeed not expected.

Reply:
That is a good ideam but it needs some tuning to work when dropping onto the last element.